### PR TITLE
USWDS - Alerts: Add alert list examples and update styles.

### LINKED ISSUE #3389: Lists inside alerts are not aligned
Screen Shot 2020-04-10 at 3 29 03 PM

### PR DESCRIPTION
**Pull request by mejiaj**
[Preview 👁](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/jm-alert-lists/components/detail/alerts--default.html)

## Description

Attemps to resolve #3389.

- Moved `margin-bottom: 0` to the last item
- Removed left padding from 1st level lists
- Removed margin-top from alert lists

## Additional information

Need confirmation on : 

- Should this behave like other lists or does it need special styling?
- Do we expect users to use `.usa-list` in their alert lists?

---

This also might be a good opportunity to update alerts.

We're adding the border via a pseudoelement [here](https://github.com/uswds/uswds/commit/c7c28fc01654b319a891200a0e10b5ce6e92314c#diff-cb5d935a33305b03898bf21efe6cb113R29-R36).

Icon and alert body are aligned via `table-cell`.

There's no real example of how `usa-alert--validation` should look like in [docs](https://designsystem.digital.gov/components/alert/).

Do we need custom alerts?

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [ ] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [ ] Run `npm test` and make sure the tests for the files you have changed have passed.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [ ] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

File: src/components/05-alerts/alerts.njk [hunk 1]
```diff
@@ -1,8 +1,110 @@
+<h6>Plain lists</h6>
 <div class="usa-alert {{ alert.classes }}" {% if alert.role %}role="{{alert.role}}"{% endif %}>
   <div class="usa-alert__body">
     {%- if alert.title %}
     <h3 class="usa-alert__heading">{{ alert.title }}</h3>
     {%- endif %}
     <p class="usa-alert__text">{{ alert.content | safe }}</p>
+    <ul>
+      <li>Seeing new patients through telehealth</li>
```
Comment (thisisdano) [inline src/components/05-alerts/alerts.njk:9]: Don't use an example tied a specific use case like telehealth

File: src/components/05-alerts/alerts.njk [hunk 2]
```diff
@@ -1,8 +1,110 @@
+<h6>Plain lists</h6>
 <div class="usa-alert {{ alert.classes }}" {% if alert.role %}role="{{alert.role}}"{% endif %}>
   <div class="usa-alert__body">
     {%- if alert.title %}
     <h3 class="usa-alert__heading">{{ alert.title }}</h3>
     {%- endif %}
     <p class="usa-alert__text">{{ alert.content | safe }}</p>
+    <ul>
```
Comment (thisisdano) [inline src/components/05-alerts/alerts.njk:8]: let's use `usa-list` for this content.

File: src/components/05-alerts/alerts.njk [hunk 3]
```diff
@@ -1,8 +1,110 @@
...
+        </ul>
+      </li>
+      <li>Integrating telehealth into your current practice</li>
+    </ul>
   </div>
-</div>
\ No newline at end of file
+</div>
+
+<h6>Nested .usa-list</h6>
```
Comment (thisisdano) [inline src/components/05-alerts/alerts.njk:37]: Do these need to be separate examples?

File: src/stylesheets/components/_alerts.scss [hunk 4]
```diff
@@ -51,14 +51,16 @@ $alert-padding-left: units($theme-alert-padding-x) +
...
   }
 
-  ul {
-    margin-bottom: 0;
-    margin-top: units(1);
-    padding-left: units(1);
-  }
-
-  .usa-checklist {
+  > ul,
```
Comment (thisisdano) [inline src/stylesheets/components/_alerts.scss:54]: let's use `usa-list` classes

**Review by thisisdano — CHANGES_REQUESTED** [review]

File: src/components/05-alerts/alerts.njk (hunk 3 above)
Comment (mejiaj) [reply src/components/05-alerts/alerts.njk]: No, it was just for testing. Removed.

**Comment by thisisdano** [comment]
These long lists can and should be in some testing example, but not in the main alerts that we'd see on the website. Those should appear as they did before.

**Review by thisisdano — APPROVED** [review]

**Pull request merged**